cmd/deploy/remote.go

```diff
@@ -191,6 +193,17 @@ func (rd *remoteDeployCommand) deploy(ctx context.Context, deployOptions *Option
 		fmt.Sprintf("%s=%d", constants.OktetoInvalidateCacheEnvVar, int(randomNumber.Int64())),
 	)
 
+	registryUrl := okteto.Context().Registry
+	subdomain := strings.TrimPrefix(registryUrl, "registry.")
+
+	ip, _, _ := net.SplitHostPort(sc.ServerName)
+	buildOptions.ExtraHosts = []types.HostMap{
+		{Hostname: registryUrl, IP: ip},
+		{Hostname: fmt.Sprintf("kubernetes.%s", subdomain), IP: ip},
+		{Hostname: fmt.Sprintf("okteto.%s", subdomain), IP: ip},
+		{Hostname: fmt.Sprintf("buildkit.%s", subdomain), IP: ip},
```

Comment:
BuildKit it is not always exposed behind our ingress controller. If BuildKit it's in its own load balancer, this wouldn't work.

I don't know/recall what information is retrieved from the backend with the cluster metadata, but we should return an specific field with the BuildKit private IP. In some configurations, it will be the same but in other cases, it will be different.

If the server already returns that IP, lets change it, if not, lets not do it as part of this PR and create another ticket for that specific scenario, so at least we fix the scenario for registry, kubernetes and API which are the most common scenario.

Reply:
I'll remove `okteto.*` (may have public override) and `buildkit.*` (may be in separate LB) for now and just keep `registry.*` and `kubernetes.*`

Reply:
Ok, I think `okteto.*` might be simpler because the CLI should have a reference to the public domain (even if it is a override) as it has to talk with it. But I'm ok with leaving it for other ticket. Mentioning it just to give context